jbpm-flow/src/main/java/org/jbpm/process/core/timer/DateTimeUtils.java

```diff
@@ -73,7 +74,24 @@ public static long parseDateAsDuration(String dateTimeStr) {
             return TimeUtils.parseTimeString(dateTimeStr);
         }
     }
-    
+
+    public static Date getDurationDate(String durationStr) {
+        // handles both period and date notation
+        try {
+            if(isPeriod(durationStr)) {
+                return (new Date(System.currentTimeMillis() +  Duration.parse(durationStr).toMillis()));
+            } else {
+                OffsetDateTime dateTime = OffsetDateTime.parse(durationStr, DateTimeFormatter.ISO_DATE_TIME);
+                Duration duration = Duration.between(OffsetDateTime.now(), dateTime);
+
+                return (new Date(System.currentTimeMillis() +  duration.toMillis()));
+            }
+        } catch(Exception e) {
+            return (new Date(System.currentTimeMillis() + TimeUtils.parseTimeString(durationStr)));
+        }
+
+    }
+
```

Comment:
Those methods are currently used by some other core classes. I would feel hesitant to change them as it would involve adding a TON more of tests for both ISO date and period format for things that this Jira does not ask for, and it might cause issues that are not related. We could do this at some later point if we want to support iso date/period format for everything but I just don't feel comfortable doing it for this jira. wdyt?

Reply:
Ah, you are right, this would implicitly enable support for things we don't want yet. :)